examples/TransactionGenerator.hs

```diff
@@ -48,344 +67,565 @@ import Servant.API
+
+transactionCommandToText :: TransactionCommand -> Text
+transactionCommandToText = T.decodeUtf8 . transactionCommandBytes
+{-# INLINE transactionCommandToText #-}
+
+transactionCommandBytes :: TransactionCommand -> B8.ByteString
+transactionCommandBytes t =
+  case t of
+    PollRequestKeys bs (MeasureTime mtime) ->
+      "poll [" <> B8.unwords bs <> "] " <> (fromString . map toLower . show $ mtime)
+    ListenerRequestKey bytestring (MeasureTime mtime) ->
+      "listen " <> bytestring <> " " <> (fromString . map toLower . show $ mtime)
+    _ -> error "impossible"
+
+transactionCommandFromText :: MonadThrow m => Text -> m TransactionCommand
+transactionCommandFromText = readTransactionCommandBytes . T.encodeUtf8
+{-# INLINE transactionCommandFromText #-}
+
+readTransactionCommandBytes :: MonadThrow m => B8.ByteString -> m TransactionCommand
+readTransactionCommandBytes = parseBytes "transaction-command" transactionCommandParser
+{-# INLINE readTransactionCommandBytes #-}
+
+transactionCommandParser :: A.Parser TransactionCommand
+transactionCommandParser = pollkeys <|> listenkeys
+
+pollkeys :: A.Parser TransactionCommand
+pollkeys = do
+  _constructor <- A.string "poll"
+  A.skipSpace
+  _open <- A.char '[' >> A.skipSpace
+  bs <- A.sepBy parseRequestKey (A.skipSpace >> A.char ',' >> A.skipSpace)
+  _close <- A.skipSpace >> A.char ']'
+  A.skipSpace
+  measure <- MeasureTime <$> ((False <$ A.string "false") <|> (True <$ A.string "true"))
+  return $ PollRequestKeys bs measure
+
+parseRequestKey :: A.Parser ByteString
+parseRequestKey = B8.pack <$> A.count 128 (A.satisfy (A.inClass "abcdef0123456789"))
+
+listenkeys :: A.Parser TransactionCommand
+listenkeys = do
+  _constructor <- A.string "listen"
+  A.skipSpace
+  bytestring <- parseRequestKey
+  A.skipSpace
+  measure <- MeasureTime <$> ((False <$ A.string "false") <|> (True <$ A.string "true"))
+  return $ ListenerRequestKey bytestring measure
+
+instance HasTextRepresentation TransactionCommand where
+  toText = transactionCommandToText
+  {-# INLINE toText #-}
+  fromText = transactionCommandFromText
+  {-# INLINE fromText #-}
+
+instance Default TransactionCommand where
+  def = RunSimpleExpressions def def
 
 data TimingDistribution
-  = Gaussian { mean :: Int
-             , var :: Int }
-  | Uniform { low :: Int
-            , high :: Int }
+  = Gaussian { mean  :: !Double , var   :: !Double }
+  | Uniform  { low   :: !Double , high  :: !Double }
   deriving (Eq, Show, Generic)
 
 instance Default TimingDistribution where
-  def = Gaussian 1000000 (div 1000000 16)
+  def = Gaussian 1000000 (1000000 / 16)
 
 instance FromJSON TimingDistribution
 
 instance ToJSON TimingDistribution
 
-data TransactionConfig = TransactionConfig
-  { _scriptCommand :: TransactionCommand
-  , _nodeChainId :: !ChainId
-  , _serverRootPrefix :: String
-  , _isChainweb :: Bool
-  , _chainwebNodePort :: !ChainwebPort
-  , _timingDistribution :: !TimingDistribution
+data ScriptConfig = ScriptConfig
```

Comment:
Minor comment on these changes to ScriptConfig and the subsequent instances (not to be addressed now)...in general I'm trending more and more away from this kind of vertical alignment.  It definitely makes things more readable, but at the cost of significantly more friction with respect to increased diff size, more merge conflicts, etc.  On individual projects this is not a big deal, but with teams it can be quite costly.